File: wowup-electron/src/app/models/curse/curse-scan-result.ts
import { AddonFolder } from "app/models/wowup/addon-folder";
import { CurseMatch } from "./curse-match";
import { CurseSearchResult } from "./curse-search-result";

export interface CurseScanResult {
    fileCount: number;
    fileDateHash?: number;
    fingerprint: number;
    folderName: string;
    individualFingerprints: number[];
    directory: string;
    addonFolder?: AddonFolder;
    exactMatch?: CurseMatch;
    searchResult?: CurseSearchResult;
}

File: WowUp.WPF/Models/Curse/CurseScanResult.cs
using WowUp.Common.Models.Curse;
using WowUp.WPF.AddonProviders.Curse;
using WowUp.WPF.Models.WowUp;

namespace WowUp.WPF.Models.Curse
{
    public class CurseScanResult
    {
        public CurseFolderScanner FolderScanner { get; set; }
        public AddonFolder AddonFolder { get; set; }
        public CurseMatch ExactMatch { get; set; }
        public CurseSearchResult SearchResult { get; set; }
    }
}
